src/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.hpp

```diff
@@ -0,0 +1,261 @@
+ * [414 (2011) 1467](http://cdsads.u-strasbg.fr/abs/2011MNRAS.414.1467P)
+ */
+class BondiHoyleAccretion {
+ public:
+  /// The mass of the black hole, \f$M\f$.
+  struct BhMass {
+    using type = double;
+    static constexpr OptionString help = {"The mass of the black hole."};
+    static type lower_bound() { return 0.0; }
+  };
+  /// The dimensionless black hole spin, \f$a_* = a/M\f$.
+  struct BhDimlessSpin {
+    using type = double;
+    static constexpr OptionString help = {"The dimensionless black hole spin."};
+    static type lower_bound() { return -1.0; }
+    static type upper_bound() { return 1.0; }
+  };
+  /// The rest mass density of the fluid far from the black hole.
+  struct RestMassDensity {
+    using type = double;
+    static constexpr OptionString help = {"The asymptotic rest mass density."};
+    static type lower_bound() { return 0.0; }
+  };
+  /// The magnitude of the spatial velocity far from the black hole.
+  struct FlowSpeed {
+    using type = double;
+    static constexpr OptionString help = {
+        "The magnitude of the asymptotic flow velocity."};
+  };
+  /// The strength of the magnetic field.
+  struct MagFieldStrength {
+    using type = double;
+    static constexpr OptionString help = {
+        "The strength of the magnetic field."};
+  };
+  /// The polytropic constant of the fluid.
+  struct PolytropicConstant {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic constant of the fluid."};
+    static type lower_bound() { return 0.0; }
+  };
+  /// The polytropic exponent of the fluid.
+  struct PolytropicExponent {
+    using type = double;
+    static constexpr OptionString help = {
+        "The polytropic exponent of the fluid."};
+    static type lower_bound() { return 1.0; }
+  };
+
+  using options =
+      tmpl::list<BhMass, BhDimlessSpin, RestMassDensity, FlowSpeed,
+                 MagFieldStrength, PolytropicConstant, PolytropicExponent>;
+
+  static constexpr OptionString help = {
+      "Axially symmetric accretion on to a Kerr black hole."};
+
+  BondiHoyleAccretion() = default;
+  BondiHoyleAccretion(const BondiHoyleAccretion& /*rhs*/) = delete;
+  BondiHoyleAccretion& operator=(const BondiHoyleAccretion& /*rhs*/) = delete;
+  BondiHoyleAccretion(BondiHoyleAccretion&& /*rhs*/) noexcept = default;
+  BondiHoyleAccretion& operator=(BondiHoyleAccretion&& /*rhs*/) noexcept =
+      default;
+  ~BondiHoyleAccretion() = default;
+
+  BondiHoyleAccretion(BhMass::type bh_mass, BhDimlessSpin::type bh_dimless_spin,
+                      RestMassDensity::type rest_mass_density,
+                      FlowSpeed::type flow_speed,
+                      MagFieldStrength::type magnetic_field_strength,
+                      PolytropicConstant::type polytropic_constant,
+                      PolytropicExponent::type polytropic_exponent) noexcept;
+
+  // @{
+  /// Retrieve hydro variable at `x`
+  template <typename DataType>
```

Comment:
Is there any reason to template on DataType for analytic data?

Reply:
@wthrowe @fmahebert since we template solutions for ray tracing/lensing your input is important :)

Reply:
My guess is that ray tracing through this analytic solution will be valuable while we learn how to make lensing visualizations of hydro in general and/or accretion flows particular.

Reply:
this is analytic data, not a solution, it's only valid at a single time

Reply:
If we can make the time-dependent geodesics read from the time-independent data, I'd say we could still make a useful check from this data. But that's getting a bit speculative, so perhaps that is best left for if/when we get to it.

Reply:
okay, then I am fine leaving this as is